diff --git a/vllm/model_executor/layers/quantization/auto_round.py b/vllm/model_executor/layers/quantization/auto_round.py
index 5d77d1e3c..17b1a24b3 100644
--- a/vllm/model_executor/layers/quantization/auto_round.py
+++ b/vllm/model_executor/layers/quantization/auto_round.py
@@ -404,6 +404,8 @@ class AutoRoundConfig(QuantizationConfig):
         return None

     def apply_ipex_quant_layer(self, layer, prefix: str):
+        from vllm.model_executor.layers.fused_moe import FusedMoE
+
         weight_bits, group_size, sym = self.get_layer_config(layer, prefix)
         if not self.check_quantized(weight_bits):
             if isinstance(layer, (LinearBase, ParallelLMHead)):
@@ -419,12 +421,22 @@ class AutoRoundConfig(QuantizationConfig):
         if isinstance(layer, (LinearBase, ParallelLMHead)):
             if "awq" in self.packing_format:
                 config = IPEXConfig(
-                    method="awq", weight_bits=weight_bits, group_size=group_size
+                    method="awq",
+                    weight_bits=weight_bits,
+                    group_size=group_size,
+                    is_qweight_sym=sym,
+                    dynamic={},
+                    full_config={},
                 )
                 return IPEXAWQLinearMethod(config)
             elif "gptq" in self.packing_format:
                 config = IPEXConfig(
-                    method="gptq", weight_bits=weight_bits, group_size=group_size
+                    method="gptq",
+                    weight_bits=weight_bits,
+                    group_size=group_size,
+                    is_qweight_sym=sym,
+                    dynamic={},
+                    full_config={},
                 )
                 return IPEXGPTQLinearMethod(config)
             else:
@@ -432,6 +444,20 @@ class AutoRoundConfig(QuantizationConfig):
                     f"ipex backend only supports awq "
                     f"and gtpq format,but got {self.packing_format}"
                 )
+        elif isinstance(layer, FusedMoE) and "gptq" in self.packing_format:
+            from vllm.model_executor.layers.quantization.ipex_quant import (
+                XPUGPTQMarlinMoEMethod,
+            )
+
+            config = IPEXConfig(
+                method="gptq",
+                weight_bits=weight_bits,
+                group_size=group_size,
+                is_qweight_sym=sym,
+                dynamic={},
+                full_config={},
+            )
+            return XPUGPTQMarlinMoEMethod(config, layer.moe_config)
         else:
             return None
