diff --git a/vllm/model_executor/models/registry.py b/vllm/model_executor/models/registry.py
--- a/vllm/model_executor/models/registry.py
+++ b/vllm/model_executor/models/registry.py
@@ -1126,7 +1126,8 @@

         input_bytes = cloudpickle.dumps((fn, output_filepath))

         returned = subprocess.run(
-            _SUBPROCESS_COMMAND, input=input_bytes, capture_output=True
+            _SUBPROCESS_COMMAND, input=input_bytes, capture_output=True,
+            env=os.environ.copy(),
         )

         # check if the subprocess is successful
