diff --git a/vllm/model_executor/models/transformers/moe.py b/vllm/model_executor/models/transformers/moe.py
index 2fa23f96f..fe17c8f80 100644
--- a/vllm/model_executor/models/transformers/moe.py
+++ b/vllm/model_executor/models/transformers/moe.py
@@ -184,7 +184,7 @@ class MoEMixin(MixtureOfExperts):

         # Positional arguments
         num_experts = self.model_config.get_num_experts()
-        top_k = getattr_iter(text_config, ["num_experts_per_tok", "top_k"], None)
+        top_k = getattr_iter(text_config, ["num_experts_per_tok", "top_k", "top_k_experts"], None)
         assert top_k is not None
         hidden_size = text_config.hidden_size
         intermediate_size = getattr_iter(
